### PR TITLE
Upgrade to powsybl dependencies 2024.2.0

### PR DESCRIPTION
**Pull request by FranckLecuyer**
Update to gridsuite/dependencies#70

### CODE REVIEW

**Review by Tristan-WorkGH — APPROVED** [review]

File: src/test/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuitTest.java [hunk 1]
```diff
@@ -29,7 +30,7 @@ class ReportMapperShortCircuitTest extends AbstractReportMapperTest {
 
     @BeforeAll
     static void prepare() throws IOException {
-        rootReportNode = RestTemplateConfig.objectMapper().readValue(AbstractReportMapperTest.class.getClassLoader().getResource("reporter_shortcircuit_test.json"), ReportNode.class);
+        rootReportNode = RestTemplateConfig.objectMapper().readValue(AbstractReportMapperTest.class.getClassLoader().getResource("reporter_shortcircuit_test.json"), ReportNodeImpl.class);
```
Comment (AbdelHedhili) [inline src/test/java/org/gridsuite/shortcircuit/server/reports/ReportMapperShortCircuitTest.java:33]: why this change ?

**Review by AbdelHedhili — APPROVED** [review]
